### ISSUE TITLE
Pyface tests failing under Traits master

### ISSUE THREAD

**Issue by corranwebster** [enthought/pyface#508]
I am getting this test failure when running the master branch of Traits:
```
======================================================================
ERROR: test_remove_missing (pyface.action.tests.test_group.TestActionItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/cwebster/.edm/envs/pyface-test-3.6-pyqt/lib/python3.6/site-packages/pyface-7.1.0.dev5-py3.6.egg/pyface/action/tests/test_group.py", line 148, in test_remove_missing
    group.remove(self.action_item)
  File "/Users/cwebster/.edm/envs/pyface-test-3.6-pyqt/lib/python3.6/site-packages/pyface-7.1.0.dev5-py3.6.egg/pyface/action/group.py", line 174, in remove
    self._items.remove(item)
  File "/Users/cwebster/.edm/envs/pyface-test-3.6-pyqt/lib/python3.6/site-packages/traits/trait_list_object.py", line 788, in remove
    self._validate_length(len(self) - 1)
  File "/Users/cwebster/.edm/envs/pyface-test-3.6-pyqt/lib/python3.6/site-packages/traits/trait_list_object.py", line 883, in _validate_length
    "element" if new_length == 1 else "elements",
traits.trait_errors.TraitError: The '_items' trait of a type instance must be a list of items which are any value, but you attempted to change its length to -1 elements.

----------------------------------------------------------------------
```

Have yet to investigate what is going on.

**Comment by kitchoi** [comment]
The test was expecting `ValueError` due to attempt to remove item from an empty list. Latest changes on `TraitListObject` validates the length and raises there, before allowing Python list to have a chance to complain.

**Comment by mdickinson** [comment]
Maybe fixed by enthought/traits#1172

**Comment by mdickinson** [comment]
Confirmed fixed by enthought/traits#1172.
